DeathsDoorComponentLogic.cs

```diff
@@ -31,7 +31,7 @@ public partial class DeathsDoorComponent {
         }
 
         public override bool Split() {
-            return remainingSplits.Count() != 0 && (SplitAIBrain() || SplitBool() || SplitScene() || SplitTruthEnding());
+            return (remainingSplits.Count() != 0 && (SplitBool() || SplitScene() || SplitTruthEnding())) || (aiBrainsToCheck.Count != 0 && SplitAIBrain());
```

Comment:
aiBrainsToCheck.count is already checked in split function
```suggestion
            return (remainingSplits.Count() != 0 && (SplitBool() || SplitScene() || SplitTruthEnding()))
                   || SplitAIBrain();
```